src/token-processor/queue/job/job.ts

```diff
@@ -43,8 +43,13 @@ export abstract class Job {
     // what to do in each case. If we choose to retry, this queue entry will simply not be marked as
     // `processed = true` so it can be picked up by the queue at a later time.
     try {
-      await this.handler();
-      status = DbJobStatus.done;
+      const success = await resolveOrTimeout(this.handler(), ENV.JOB_QUEUE_TIMEOUT_MS);
```

Comment:
When this times out and returns continues with `success==false`, what happens to the underlying promise? It doesn't look like there's termination code?

Reply:
That part is supposed to be handled by the tookit here: https://github.com/hirosystems/api-toolkit/blob/develop/src/helpers/time.ts#L68
But let me know if you think I should add something extra like an abort controller, etc.

Reply:
Here's the scenario that seems like it could cause issues:
```js
async function someTask() {
  await sleep(10_000);
  console.log('something happens because task was still going');
}
const success = await resolveOrTimeout(someTask(), 100);
console.log(`Task done, success: ${success}`);
```

In something like the above, you'd hit the `task done`, then later on have a dangling side-effect triggered in the background